src/catalog/src/kvbackend/manager.rs

```diff
@@ -320,12 +345,21 @@ struct SystemCatalog {
 }
 
 impl SystemCatalog {
-    // TODO(j0hn50n133): remove the duplicated hard-coded table names logic
-    fn schema_names(&self) -> Vec<String> {
-        vec![
-            INFORMATION_SCHEMA_NAME.to_string(),
-            PG_CATALOG_NAME.to_string(),
-        ]
+    fn schema_names(&self, catalog_protocol: CatalogProtocol) -> Vec<String> {
+        match catalog_protocol {
+            // pg_catalog only visible under postgres protocol
+            CatalogProtocol::PostgreSQL => vec![PG_CATALOG_NAME.to_string()],
+            // information_schema visible for mysql protocol and other
+            CatalogProtocol::MySQL => {
+                vec![INFORMATION_SCHEMA_NAME.to_string()]
+            }
+            CatalogProtocol::Other => {
+                vec![
+                    PG_CATALOG_NAME.to_string(),
+                    INFORMATION_SCHEMA_NAME.to_string(),
+                ]
+            }
```

Comment:
`information_schema` is needed for both postgres and mysql. This logic can be changed to appending `pg_catalog` when it's calling from postgres